cmake: sca: Enable CodeChecker error exit status

noahp opened the pull request:
Normally the return code of `CodeChecker analyze` and `CodeChecker parse`
is suppressed, so all the enabled commands can execute instead of
crashing the build.

Add a new option, `CODECHECKER_PARSE_EXIT_STATUS`, to permit failing the
build if `CodeChecker parse` returns non-zero.

Signed-off-by: Noah Pendleton <noah.pendleton@gmail.com>

pdgendt commented:
I don't think this is a "clean" solution, you could create a variable for the `parse` command and simply not append
```cmake
|| ${CMAKE_COMMAND} -E true
```
in case your variable exists.

noahp committed:
cmake: sca: Enable CodeChecker error exit status

Normally the return code of `CodeChecker analyze` and `CodeChecker parse`
is suppressed, so all the enabled commands can execute instead of
crashing the build.

Add a new option, `CODECHECKER_PARSE_EXIT_STATUS`, to permit failing the
build if `CodeChecker parse` returns non-zero.

Signed-off-by: Noah Pendleton <noah.pendleton@gmail.com>

noahp commented:
> I don't think this is a "clean" solution, you could create a variable for the `parse` command and simply not append
> 
> ```cmake
> || ${CMAKE_COMMAND} -E true
> ```
> 
> in case your variable exists.

Great idea, changed it to use that!
